cls_module/cls_module/memory/stm/aha.py

```diff
@@ -231,7 +299,26 @@ def forward_pr(self, inputs, targets):
 
   def build_pc(self, pc_config, pc_input_shape):
     """Builds the Pattern Completion (PC) module."""
-    del pc_config
+
+    if self.is_hebbian_perforant():
+      # Builds single set of weights (doesn't currently exist, because activations are copied across)
+      # PC must be initialised with random weights
+      ec_size = np.prod(self.input_shape[1:])
+      ps_size = np.prod(pc_input_shape[1:])
+
+      dg_ca3 = LocalConnection(ps_size, ps_size, bias=False).to(self.device)
+      dg_ca3_optimizer = LocalOptim(dg_ca3.named_parameters(), lr=pc_config['learning_rate'])
+
+      self.add_module('dg_ca3', dg_ca3)
+      self.add_optimizer('dg_ca3', dg_ca3_optimizer)
+
+      ec_ca3 = LocalConnection(ec_size, ps_size, bias=False).to(self.device)
+      ec_ca3_optimizer = LocalOptim(ec_ca3.named_parameters(), lr=pc_config['learning_rate'])
+
+      self.add_module('ec_ca3', ec_ca3)
+      self.add_optimizer('ec_ca3', ec_ca3_optimizer)
+
+      self.learning_rule = PureHebbRule()
 
     # Initialise the buffer
     self.pc_buffer = None
```

Comment:
Yeah, because the `buffer` is really initialised when the first inputs come in. Aside from that, there are no layers to build and nothing else to initialise.

I guess technically, rather than `self.pc_buffer = None`, you could initialise it with 'zeros'.. but then that will just get replaced when the first step hits `forward_pc`.